com.sap.adt.abapcleaner/src/com/sap/adt/abapcleaner/programbase/CommandLineArgs.java

```diff
@@ -128,11 +145,15 @@ public static CommandLineArgs create(Persistency persistency, String[] args) {
 				++i;
 		}
 		
+		if (!StringUtil.isNullOrEmpty(sourceFolder)) {
+			sourceFiles = persistency.getFilesInDirectory(sourceFolder, StringUtil.isNullOrEmpty(fileFilter) ? "*.abap" : fileFilter, true);
+		}
+		
 		if (!overwrite && !StringUtil.isNullOrEmpty(targetPath) && persistency.fileExists(targetPath)) {
 			errors.append("Target file already exists; please use " + OPT_OVERWRITE + " to allow overwriting: " + targetPath).append(LINE_SEP);
 		}
 		
-		return new CommandLineArgs(sourceCode, cleanupRange, profileData, abapRelease, targetPath, overwrite, partialResult, showStats, showUsedRules, errors.toString(), showHelp);
+		return new CommandLineArgs(sourceCode, cleanupRange, profileData, abapRelease, targetPath, sourceFiles, overwrite, partialResult, showStats, showUsedRules, errors.toString(), showHelp);
 	}
 
 	public static String getHelp(Persistency persistency) {
```

Comment:
Please enhance getHelp() for the new options. I guess the beginning should be something like:
{ sourcedir [filepattern] } / { { sourcefile / sourcecode } [linerange] }
and then further down, we could have 
[ {targetdir / targetfile} [overwrite] ]
And an explanation of the new options "sourcedir" and "filepattern".